apps/portals/src/configurations/elportal/synapseConfigs/handleFilesToParticipants.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  ColumnSingleValueFilterOperator,
+  ColumnSingleValueQueryFilter,
+} from '@sage-bionetworks/synapse-types'
+import { SynapseClient, SynapseUtilityFunctions } from 'synapse-react-client'
+import { CustomControlCallbackData } from 'synapse-react-client/src/components/SynapseTable/TopLevelControls/TopLevelControls'
+
+export const handleFilesToParticipants = async (
+  event: CustomControlCallbackData,
+) => {
+  // add filter for files perspective, to show participants associated to the selected files only.
+  const idColIndex = event.data?.columnModels?.findIndex(
+    (cm) => cm.name === 'fileId',
+  )
+
+  const token = await SynapseClient.getAccessTokenFromCookie()
+  const queryResultBundle = await SynapseClient.getFullQueryTableResults(
+    event.request!,
```

Comment:
Strongly recommend overriding partMask to be 1 (row data only), unless that happens in an earlier step